src/common/hostPlatform.ts

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import {ChildProcess} from "./node/childProcess";
+import * as path from "path";
+import * as Q from "q";
+
+/**
+ * Interface defining the host (desktop) platform specific operations.
+ */
+interface IHostPlatform {
+    getUserHomePath(): string;
+    getSettingsHome(): string;
+    getCommand(packageName: string): string;
```

Comment:
Nit: Calling this the "packageName" is a bit misleading, especially since we are passing "react-native" as the argument whereas the actual package name that exposes the command is "react-native-cli". May be "cliName" ?